projects/proto/simulation.proto

```diff
@@ -1,149 +1,97 @@
+message pbSimulation {
+    pbSensor sensors = 1;
+    pbActuator actuators = 2;
 
-    optional SensorMessage sensors =                        3;
-    optional ActuatorMessage actuators =                    4;
+    pbModelStateSpace model = 3;
+    pbModelState state = 4;
+    pbModelInput input = 5;
+    pbModelAuxiliaryState auxiliary_state = 6;
 
-    optional BicycleStateMessage state =                    5;
-    optional BicycleInputMessage input =                    6;
-    optional BicyclePoseMessage pose =                      7;
+    pbController controller = 7;
 
-    optional BicycleModelMessage model =                    8;
-    optional BicycleKalmanMessage kalman =                  9;
-
-    optional BicycleAuxiliaryStateMessage auxiliary_state = 10;
-
-    optional TimingMessage timing =                         11;
-
-    optional float feedback_torque =                        12;
-
-    optional ControllerMessage controller =                 13;
-}
-
-message FirmwareVersion {
-    required bytes f = 1;
+    pbTiming timing = 8;
 }
 
 // states are yaw, roll, steer, roll rate, steer rate (5)
-message BicycleStateMessage {
+message pbModelState {
     repeated float x = 1;
 }
 
 // auxiliary states are x, y, rear wheel angle, pitch angle (4)
-message BicycleAuxiliaryStateMessage {
+message pbModelAuxiliaryState {
     repeated float x = 1;
 }
 
 // inputs are roll torque, steer torque (2)
-message BicycleInputMessage {
+message pbModelInput {
     repeated float u = 1;
 }
 
 // STM32F4 ADC, DAC are 12-bit
 // maximum rear wheel encoder count is 1152
-message SensorMessage {
-    required uint32 kistler_measured_torque =   1;
-    required uint32 kollmorgen_actual_torque =  2;
-    required uint32 steer_encoder_count =       3;
-    optional uint32 rear_wheel_encoder_count =  4;
+message pbSensor {
+    uint32 kistler_measured_torque = 1;
+    uint32 kollmorgen_measured_torque = 2;
+    uint32 steer_encoder_count = 3;
+    uint32 rear_wheel_encoder_count = 4;
 }
 
-message ActuatorMessage {
-    required uint32 kollmorgen_command_torque = 1;
-}
-
-// symmetric 5x5 matrix
-message SymmetricStateMatrixMessage {
-    repeated float m = 1;
-}
-
-// symmetric 2x2 matrix
-message SymmetricOutputMatrixMessage {
-    repeated float m = 1;
-}
-
-// 5x2 matrix
-message KalmanGainMatrixMessage {
-    repeated float m = 1;
-}
-
-message BicycleKalmanMessage {
-    optional BicycleStateMessage state_estimate =                          1;
-    optional SymmetricStateMatrixMessage error_covariance =                2;
-    optional SymmetricStateMatrixMessage process_noise_covariance =        3;
-    optional SymmetricOutputMatrixMessage measurement_noise_covariance =   4;
-    optional KalmanGainMatrixMessage kalman_gain =                         5;
+message pbActuator {
+    uint32 kollmorgen_command_torque = 1;
 }
 
 // 2x2 matrix
-message SecondOrderMatrixMessage {
+message pbSecondOrderMatrix {
     repeated float m = 1;
```

Comment:
Can decrease the size of all matrix messages by adding the option ```(nanopb).fixed_length = true``` which drops the repeat count value.

Reply:
Looks like it only works for bytes: https://github.com/nanopb/nanopb/issues/260